main/src/main/scala/sbt/internal/TaskProgress.scala

```diff
@@ -83,41 +84,40 @@ private[sbt] final class TaskProgress(log: ManagedLogger)
       case _ =>
     }
   }
+  private[this] def appendProgress(event: ProgressEvent): Unit = {
+    import sbt.internal.util.codec.JsonProtocol._
+    log.logEvent(Level.Info, event)
+  }
   private[this] def active: Vector[Task[_]] = activeTasks.toVector.filterNot(Def.isDummy)
+  private[this] def activeExceedingThreshold: Vector[(Task[_], Long)] = active.flatMap { task =>
+    val elapsed = timings.get(task).currentElapsedMicros
+    if (elapsed.micros > threshold) Some[(Task[_], Long)](task -> elapsed) else None
```

Comment:
👍. Initially this looked confusing until I realized that the comparison of microsecond and millisecond is done using Duration.